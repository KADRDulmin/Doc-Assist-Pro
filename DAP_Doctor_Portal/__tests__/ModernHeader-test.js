import React from 'react';
import { render } from '@testing-library/react-native';
import { Text, View } from 'react-native';

// A simplified mock of ModernHeader for testing
const MockModernHeader = ({ title = 'Doc Assist Pro' }) => (
  <View testID="header-container">
    <Text testID="header-title">{title}</Text>
  </View>
);

describe('ModernHeader Component Tests', () => {
  it('renders with default title', () => {
    const { getByTestId } = render(<MockModernHeader />);
    const titleElement = getByTestId('header-title');
    expect(titleElement.props.children).toBe('Doc Assist Pro');
  });

  it('renders with custom title', () => {
    const { getByTestId } = render(<MockModernHeader title="Patient Dashboard" />);
    const titleElement = getByTestId('header-title');
    expect(titleElement.props.children).toBe('Patient Dashboard');
  });

  it('passes snapshot test', () => {
    const tree = render(<MockModernHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
