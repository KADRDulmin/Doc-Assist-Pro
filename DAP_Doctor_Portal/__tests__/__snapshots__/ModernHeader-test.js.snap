// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModernHeader Component Tests passes snapshot test 1`] = `
<View
  testID="header-container"
>
  <Text
    testID="header-title"
  >
    Doc Assist Pro
  </Text>
</View>
`;
